### PR TITLE
Add a kustomize manifest for deploying odh-dashboard

### PR DESCRIPTION
**Pull request by LaVLaS**
This updates the `kustomize` manifest so that can be used to deploy odh-dashboard directly or included in a kfdef for deployment.

Manual deployment & deletion works with `make deploy` and `make undeploy`.  You can also run `kustomize build manifests/overlays/authentication | oc apply -f -`.

If you want to test that this works with a kfdef you can add a `repos` for this PR and reference the manifest name
```
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  name: opendatahub
spec:
  applications:
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: odh-common
    name: odh-common
  - kustomizeConfig:
      parameters:
        - name: s3_endpoint_url
          value: "s3.odh.com"
      repoRef:
        name: manifests
        path: jupyterhub/jupyterhub
    name: jupyterhub
  - kustomizeConfig:
      overlays:
      #- cuda-11.0.3
      - additional
      repoRef:
        name: manifests
        path: jupyterhub/notebook-images
    name: notebook-images
  - kustomizeConfig:
      repoRef:
        name: manifests-dashboard
        path: odh-dashboard
    name: odh-dashboard
  repos:
  - name: manifests
    uri: https://github.com/opendatahub-io/odh-manifests/tarball/master
  - name: manifests-dashboard
    uri: https://github.com/opendatahub-io/odh-dashboard/tarball/pull/222/head
```

### CODE REVIEW

**Review by dlabaj — APPROVED** [review]
LGTM

**Comment by dlabaj** [comment]
@LaVLaS  Should this go to main instead of master?

**Comment by LaVLaS** [comment]
> @LaVLaS Should this go to main instead of master?

Old habit.  Updated to hit `main`.  I still need to check this against #211 to make sure that it has all of the required permissions

File: manifests/base/kustomization.yaml [hunk 1]
```diff
@@ -12,3 +12,7 @@ resources:
 - deployment.yaml
 - routes.yaml
 - service.yaml
+images:
+  - name: odh-dashboard
+    newName: quay.io/opendatahub/odh-dashboard
+    newTag: v1.0
```
Comment (lucferbux) [inline manifests/base/kustomization.yaml:18]: Despite the fact that the tag will change once we have a new image, right now it seems that the latest version was v1.0.0.1, not sure if there are some significant changes between versions.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**